project_stock/models/project_task.py

```diff
@@ -104,14 +105,35 @@ def onchange_picking_type(self):
         self.location_id = self.picking_type_id.default_location_src_id.id
         self.location_dest_id = self.picking_type_id.default_location_dest_id.id
 
-    @api.onchange("location_id", "move_ids")
-    def _onchange_location(self):
-        self.move_ids.update(
+    def _check_tasks_with_pending_moves(self):
+        if self.move_ids and "assigned" in self.mapped("move_ids.state"):
+            raise UserError(
+                _("It is not possible to change this with reserved movements in tasks.")
+            )
+
+    @api.onchange("location_id")
+    def onchange_location_id(self):
+        self._check_tasks_with_pending_moves()
+
+    @api.onchange("location_dest_id")
+    def onchange_location_dest_id(self):
+        self._check_tasks_with_pending_moves()
+
+    def _update_moves_info(self):
+        self._check_tasks_with_pending_moves()
+        picking_type = self.picking_type_id or self.project_id.picking_type_id
```

Comment:
This is `self.ensure_one()` right now, but it's being called for several records.